ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/slot/DiskSlots.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2020, Red Hat, Inc. and/or its affiliates,
+ * and individual contributors as indicated by the @author tags.
+ * See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ * This copyrighted material is made available to anyone wishing to use,
+ * modify, copy, or redistribute it subject to the terms and conditions
+ * of the GNU Lesser General Public License, v. 2.1.
+ * This program is distributed in the hope that it will be useful, but WITHOUT A
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2020,
+ * @author Jonathan Halliday (jonathan.halliday@redhat.com)
+ */
+package com.arjuna.ats.internal.arjuna.objectstore.slot;
+
+import com.arjuna.ats.arjuna.logging.tsLogger;
+
+import java.io.*;
+import java.util.zip.CRC32;
+
+/**
+ * File backed implementation of the SlotStore backend.
+ * WARNING: this is currently a test implementation and is NOT SAFE for production use yet.
```

Comment:
The current code shouldn't be too bad, but I'm not going to be really comfortable until we've done a run over the full test suite with a config to use that store as the backend. So far the only test coverage is small subset that I ran locally.

For serious production use, the main challenge will be how to deal with changing config values. For example, if you reduce the number of slots whilst the store is non-empty, some records may become inaccessible. Likewise bad things can happen if you change the slot size.  I'm not sure these need code safeguards, it may be sufficient just to update the docs to reflect the potential problems and put a big 'don't do this' notice in.

Reply:
Thanks